### ISSUE TITLE
[front] feat: Language filter for recommendation language should propose all languages.

### ISSUE THREAD

**Issue by lfaucon** [tournesol-app/tournesol#627]
Currently only languages French, English and German are available for selection, but there is no good reason and it makes Tournesol unwelcoming for contributors with a different language.

We should enable more languages by using a multi-select component https://mui.com/api/multi-select-unstyled/#main-content

**Comment by sigmike** [comment]
What language list should we use? The [ISO 639-1 codes](https://en.wikipedia.org/wiki/List_of_ISO_639-1_codes)?

**Comment by GresilleSiffle** [comment]
Good question : )

In the back end we currently use a subset of languages which represent all languages that Django provides a translation for.

See the current list ( directly from the Django source code ):
https://github.com/django/django/blob/e0442a628eb480eac6a7888aed5a86f83499e299/django/conf/global_settings.py#L56

When the back end fetch the video metadata, it will store the language in the database only if it matches one language of this list. 
See:
https://github.com/tournesol-app/tournesol/blob/8f86cecca830d4109757096a7e74aaeda9a14226/backend/tournesol/utils/video_language.py#L51

Do you think it's a good idea to use theses languages @amatissart @lfaucon ?

**Comment by lfaucon** [comment]
Yes, it's good to use Django's `settings.LANGUAGES` with a good way to get this list to the frontend.

**Comment by sigmike** [comment]
We also need the list in the browser extension to filter the languages from the browser[[1]](https://github.com/tournesol-app/tournesol/blob/1028d98feea29f841e552ad79748afaa112087cc/browser-extension/background.js#L102). But if we accept all the languages then maybe we can just remove this filter. How would the backend behave if the recommendations request includes languages it doesn't know?

> with a good way to get this list to the frontend

What is a good way to do that? A new API call?

**Comment by amatissart** [comment]
After our discussion last Friday, it seems that 2 approaches may be used to provide a list of available languages both in the frontend and the extension:

1. Generate a static JSON file, using the list internally by `langdetect`. Here is a way to build this list programmatically:

```python
from langdetect import detector_factory
detector_factory.init_factory()
detector_factory._factory.get_lang_list()
```

2. Implement a new API endpoint to expose the list of languages for which videos actually exist in the database.
Advantage: the list would be shorter and the options would be more likely to return actual recommendations
Drawback: one more asynchronous call may be required in frontend and extension before recommendations can be loaded.

Either approach seems reasonable to me.

In any case, I think it's reasonable to ignore the language variants, and focus on 2 or 3-letters language code (ISO 639-1, ISO 639-3). If edge cases about scripts variants remain, feel free to ignore them for now, and possibly open a new issue to describe these cases with more details.

**Comment by sigmike** [comment]
Actually I switched back to a constant in the code to keep things simple because the generating script started becoming a little complex (we don't use all langdetect's list and Django doesn't provide a translation for all the languages). It removes complexity in updating the language list and make adding a new interface language much simpler (all translations at the same place).

I only removed Chinese from the list imported from langdetect because it's not clear how to handle the subtags. Langdetect will return `zh-cn` or `zh-tw` on some videos though, so for now these video can only be recommended when all the languages are unselected from the filter.
